pyuvdata/uvbeam/beamfits.py

```diff
@@ -298,41 +298,8 @@ def read_beamfits(
                 self.history, self.pyuvdata_version_str
             ):
                 self.history += self.pyuvdata_version_str
-            while "HISTORY" in primary_header.keys():
-                primary_header.remove("HISTORY")
-
-            # remove standard FITS header items that are still around
-            std_fits_substrings = [
-                "SIMPLE",
-                "BITPIX",
-                "EXTEND",
-                "BLOCKED",
-                "GROUPS",
-                "PCOUNT",
-                "BSCALE",
-                "BZERO",
-                "NAXIS",
-                "PTYPE",
-                "PSCAL",
-                "PZERO",
-                "CTYPE",
-                "CRVAL",
-                "CRPIX",
-                "CDELT",
-                "CROTA",
-                "CUNIT",
-            ]
-            for key in list(primary_header.keys()):
-                for sub in std_fits_substrings:
-                    if key.find(sub) > -1:
-                        primary_header.remove(key)
-
-            # find all the remaining header items and keep them as extra_keywords
-            for key in primary_header:
-                if key == "COMMENT":
-                    self.extra_keywords[key] = str(primary_header.get(key))
-                elif key != "":
-                    self.extra_keywords[key] = primary_header.get(key)
+
+            self.extra_keywords = uvutils._get_fits_extra_keywords(primary_header)
```

Comment:
Is there any circumstance in which self.extra_keywords would already have entries? In which case, shouldn't we use update?

Also, does it matter that self.extra_keywords is actually a UVParameter and not a dictionary?

Reply:
I don't think it should have entries already, this is in the code that reads a single file.

Actually `self.extra_keywords` is a property linked to `self._extra_keywords` which is the UVParameter. This is the right way to set it.